message_app_server/message_app/send_message.py

```diff
@@ -7,7 +7,7 @@
 from PIL import Image
 
 send_messages = Blueprint("send_messages", __name__)
-socketio = SocketIO(cors_allowed_origins='*')
+socketio = SocketIO(cors_allowed_origins=['http://localhost:3000', 'http://localhost:5000'])
```

Comment:
Is there any difference between setting the CORS here and this?
`CORS(app, origins=["http://localhost:3000"])`

Reply:
Yes. `CORS(app, origins=["http://localhost:3000"])` is for the client fetching data through API, while this one is for WebSocket. They are not related to each other so we have to set up cors for both.

I passed `'http://localhost:5000'` to the whitelist so the old version can still work. It will be deleted after we finish migrating.

Reply:
What do you mean by old version? Old version of WebSocket?

Also, please add a note here about what you just said
`// whitelist 'http://localhost:5000' to keep old WebSocket works. It will be deleted after we finish migrating`

Reply:
The one using plain javascript (the file `socketio.js`). Its domain is `http://localhost:5000`, so I have to put that domain into the whitelist to make that version works.

Reply:
Gotcha. thanks.